src/python/dxpy/scripts/dx.py

```diff
@@ -3249,15 +3249,26 @@ def print_run_input_help():
 def run(args):
     if args.help:
         print_run_help(args.executable, args.alias)
-
+    client_ip = None
     if args.allow_ssh is not None:
-        args.allow_ssh = [i for i in args.allow_ssh if i is not None]
-    if args.allow_ssh == [] or ((args.ssh or args.debug_on) and not args.allow_ssh):
-        args.allow_ssh = ['*']
+        for i, ip in enumerate(args.allow_ssh):
+            if ip is None:
+                if client_ip is not None:
+                    del args.allow_ssh[i]
```

Comment:
Is it safe to delete from list while iterating through it?

could we iterate over something like this instead?
```
[ip for ip in args.allow_ssh if not (ip is None and client_ip is not None)]
```

Also, why do we delete `args.allow_ssh[i]` if client_ip is known, but if it isnt, we go get it and then set `args.allow_ssh[i]` to the value?